### DISCUSSION TITLE
SchemaGeneratorHooks not loaded via Spring Autowiring (UUID parsing)

### DISCUSSION THREAD

**Discussion by psud** [ExpediaGroup/graphql-kotlin#1545] [Q&A] [answered]
**Context**
I would like to pass UUIDs into the graphql schemas. As described in [the scalar docs](https://opensource.expediagroup.com/graphql-kotlin/docs/schema-generator/writing-schemas/scalars/), this is not possible by default. 
The solution is described in the same article. I copy&pasted the `class CustomSchemaGeneratorHooks : SchemaGeneratorHooks` next to my primary schema in the same module and same package. Additionally, I gave it an `@Component` annotation to be picked up by Spring autowiring.

**Problem**
When starting up the application I receive following error
`
Caused by: com.expediagroup.graphql.generator.exceptions.TypeNotSupportedException: Cannot convert java.util.UUID since it is not a valid GraphQL type or outside the supported packages "[com.travacado.mono]"
`

**Question**
How do I do the step described in the documentation: "Custom scalars have to be explicitly added to the schema through SchemaGeneratorHooks.willGenerateGraphQLType". How do I tell Spring to use apply the `SchemaGeneratorHook` to the `GeneratorConfiguration`?

**Relevant Code**
```
@GraphQLDescription("My schema description")
@Component
class MySchema : Schema

@Component
class TestService : Query {
    fun test(id: UUID): GqlChecklist {
        return GqlChecklist(ChecklistId(id), "test", emptyList())
    }
}
```

```
@Component
class CustomSchemaGeneratorHooks : SchemaGeneratorHooks {

    override fun willGenerateGraphQLType(type: KType): GraphQLType? = when (type.classifier as? KClass<*>) {
        UUID::class -> graphqlUUIDType
        else -> null
    }
}

val graphqlUUIDType: GraphQLScalarType = GraphQLScalarType.newScalar()
    .name("UUID")
    .description("A type representing a formatted java.util.UUID")
    .coercing(UUIDCoercing)
    .build()

object UUIDCoercing : Coercing<UUID, String> {
    ...
}
```

**Final Stacktrace Message** (contains all messages from propagated causes)
`org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'webHandler' defined in class path resource [org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration$EnableWebFluxConfiguration.class]: Initialization of bean failed; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'routerFunctionMapping' defined in class path resource [org/springframework/boot/autoconfigure/web/reactive/WebFluxAutoConfiguration$EnableWebFluxConfiguration.class]: Invocation of init method failed; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'com.expediagroup.graphql.server.spring.GraphQLRoutesConfiguration': Unsatisfied dependency expressed through constructor parameter 1; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'springGraphQLServer' defined in class path resource [com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.class]: Unsatisfied dependency expressed through method 'springGraphQLServer' parameter 2; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'graphQLRequestHandler' defined in class path resource [com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.class]: Unsatisfied dependency expressed through method 'graphQLRequestHandler' parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'graphQL' defined in class path resource [com/expediagroup/graphql/server/spring/GraphQLSchemaConfiguration.class]: Unsatisfied dependency expressed through method 'graphQL' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'schema' defined in class path resource [com/expediagroup/graphql/server/spring/FederatedSchemaAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [graphql.schema.GraphQLSchema]: Factory method 'schema' threw exception; nested exception is com.expediagroup.graphql.generator.exceptions.TypeNotSupportedException: Cannot convert java.util.UUID since it is not a valid GraphQL type or outside the supported packages "[com.travacado.mono]"`

Thank you!

**Comment by dariuszkuc** [comment, marked as answer]
Your setup looks correct. Do you have a link to a repo that reproduces this issue? You can also take a look at our [example app](https://github.com/ExpediaGroup/graphql-kotlin/tree/master/examples/server/spring-server) setup.

**Reply by psud** [reply]
Well... I created a minimum implementation to share with you and copied files from my source project. Everything worked great.
Then I went back to the source project and to my amazement it worked, too. No clue what was going on, I cleaned caches beforehand and even restarted the computer but something must have changed.
Anyways, thank you @dariuszkuc ! Also, the example apps are amazing - Awesome work!
